### PR TITLE
feat: remove AuthenticationFilter in favor of SecurityPolicy

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR removes `AuthenticationFilter`. The existing JWT authentication in `AuthenticationFilter` has been replaced by `SecurityPolicy`.  The other planned authn features (OIDC, Basic Auth, etc.) will be incorporated into `SecurityPolicy as well in the future.

Related: https://github.com/envoyproxy/gateway/pull/2079

### CODE REVIEW

**Comment by zhaohuabing** [comment]
The code is ready for review while I'm working on the docs.

**Comment by zirain** [comment]
kindly reminder: you need rebase once https://github.com/envoyproxy/gateway/pull/2081 merged.

**Comment by arkodg** [comment]
@zhaohuabing you will also need to delete and recreate the `crds`  in https://github.com/envoyproxy/gateway/tree/main/charts/gateway-helm since our tooling does not clean, only adds

**Review by zirain — APPROVED** [review]
LGTM

**Comment by Xunzhuo** [comment]
> @zhaohuabing you will also need to delete and recreate the `crds` in https://github.com/envoyproxy/gateway/tree/main/charts/gateway-helm since our tooling does not clean, only adds

How about raising an issue for it ? @arkodg

**Review by Xunzhuo — COMMENTED** [review]
Well done, some docs lint errors need to be resolved.

File: site/content/en/latest/user/jwt-authentication.md [hunk 1]
```diff
@@ -1,66 +1,71 @@
...
+title: "JWT Authentication"
 ---
 
 This guide provides instructions for configuring [JSON Web Token (JWT)][jwt] authentication. JWT authentication checks
 if an incoming request has a valid JWT before routing the request to a backend service. Currently, Envoy Gateway only
 supports validating a JWT from an HTTP header, e.g. `Authorization: Bearer <token>`.
 
-## Installation
+Envoy Gateway introduces a new CRD called [SecurityPolicy][] that allows the user to configure JWT authentication. 
+This instantiated resource can be linked to a [Gateway][], [HTTPRoute][] or [GRPCRoute][] resource.
```
Comment (Xunzhuo) [inline site/content/en/latest/user/jwt-authentication.md:10]: we should put the refer short name in `[]`

File: site/content/en/latest/user/jwt-authentication.md [hunk 2]
```diff
@@ -1,66 +1,71 @@
...
+## Prerequisites
 
 Follow the steps from the [Quickstart](quickstart.md) guide to install Envoy Gateway and the example manifest.
 For GRPC - follow the steps from the [GRPC Routing](grpc-routing.md) example.
 Before proceeding, you should be able to query the example backend using HTTP or GRPC.
 
 ## Configuration
 
-Allow requests with a valid JWT by creating an [AuthenticationFilter][] and referencing it from the example HTTPRoute or GRPCRoute.
+Allow requests with a valid JWT by creating an [SecurityPolicy][] and attaching it to the example HTTPRoute or GRPCRoute.
```
Comment (Xunzhuo) [inline site/content/en/latest/user/jwt-authentication.md:20]: ditto

**Comment by zhaohuabing** [comment]
This PR only adds docs for JWT, cors docs will be taken care of in a follow-up PR.

**Comment by zhaohuabing** [comment]
@zirain @Xunzhuo @arkodg please hold other PRs until this one get merged. It's a bit hard to rebase such a huge PR.

**Comment by zhaohuabing** [comment]
Got a DATA RACE error running the test in Github workflow, but can't reproduce on my local machine with ` make go.test.coverage`, any idea?

https://github.com/envoyproxy/gateway/actions/runs/6665529782/job/18115878813?pr=2086

**Comment by zhaohuabing** [comment]
> Got a DATA RACE error throw from the Kubernetes lib when running the test in Github workflow, but can't reproduce on my local machine with ` make go.test.coverage`, any idea?
> 
> https://github.com/envoyproxy/gateway/actions/runs/6665529782/job/18115878813?pr=2086

The tests passed after a few retries. Looks like bugs in fake client.

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**